numpy/core/src/multiarray/mapping.c

```diff
@@ -2567,6 +2563,28 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     NPY_END_THREADS;
     Py_DECREF(intp_type);
     return 0;
+
+indexing_error:
+
+    if (mit->size == 0) {
+        PyObject *err_type = NULL, *err_value = NULL, *err_traceback = NULL;
+        PyErr_Fetch(&err_type, &err_value, &err_traceback);
+        /* 2020-04-02, NumPy 1.19 */
+        if (DEPRECATE(
+                "Invalid index into empty indexing result found. "
+                "Such indexing errors will not be ignored in the future. "
+                "(Please raise this DeprecationWarning for error details "
+                "about the incorrect index.)") < 0) {
```

Comment:
Perhaps you should mention `warninga.simplefilter("error")` here, this wording is confusing.

Reply:
Right, tried to rephrase, could also leave it out, or we could start to add that to most of our deprecation warnings, heh... I did not test the full test suit (because mistake I had first), so updated for that. In theory `NpyIter_Dealloc` can error return,  but it is low priority, and I did not want to add it to this PR.